packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraSettingsTests.m

```diff
@@ -0,0 +1,256 @@
+  OCMStub([writerMock startWriting]).andDo(^(NSInvocation *invocation) {
+    status = AVAssetWriterStatusWriting;
+  });
+  OCMStub([writerMock status]).andDo(^(NSInvocation *invocation) {
+    [invocation setReturnValue:&status];
+  });
+
+  // Expect FPS configuration is passed to camera device.
+  [self waitForExpectations:@[
+    lockExpectation, beginConfigurationExpectation, minFrameDurationExpectation,
+    maxFrameDurationExpectation, commitConfigurationExpectation, unlockExpectation
+  ]
+                    timeout:1
+               enforceOrder:YES];
+
+  id videoMock = OCMClassMock([AVAssetWriterInputPixelBufferAdaptor class]);
+  OCMStub([videoMock assetWriterInputPixelBufferAdaptorWithAssetWriterInput:OCMOCK_ANY
+                                                sourcePixelBufferAttributes:OCMOCK_ANY])
+      .andReturn(videoMock);
+
+  id writerInputMock = [OCMockObject niceMockForClass:[AVAssetWriterInput class]];
+
+  // Expect audio bitrate is passed to writer.
+  XCTestExpectation *audioSettingsExpectation =
+      [self expectationWithDescription:@"audioSettingsExpectation"];
+
+  OCMStub([writerInputMock assetWriterInputWithMediaType:AVMediaTypeAudio
+                                          outputSettings:[OCMArg any]])
+      .andDo(^(NSInvocation *invocation) {
+        NSMutableDictionary *args;
+        [invocation getArgument:&args atIndex:3];
+
+        if ([args[AVEncoderBitRateKey] isEqual:@(TEST_AUDIO_BITRATE)]) {
+          [audioSettingsExpectation fulfill];
+        }
+      })
+      .andReturn(writerInputMock);
+
+  // Expect FPS and video bitrate are passed to writer.
+  XCTestExpectation *videoSettingsExpectation =
+      [self expectationWithDescription:@"videoSettingsExpectation"];
+
+  OCMStub([writerInputMock assetWriterInputWithMediaType:AVMediaTypeVideo
+                                          outputSettings:[OCMArg any]])
+      .andDo(^(NSInvocation *invocation) {
+        NSMutableDictionary *args;
+        [invocation getArgument:&args atIndex:3];
+
+        if ([args[AVVideoCompressionPropertiesKey][AVVideoAverageBitRateKey]
+                isEqual:@(TEST_VIDEO_BITRATE)] &&
+            [args[AVVideoCompressionPropertiesKey][AVVideoExpectedSourceFrameRateKey]
+                isEqual:@(TEST_FPS)]) {
+          [videoSettingsExpectation fulfill];
+        }
+      })
+      .andReturn(writerInputMock);
+
+  FLTThreadSafeFlutterResult *result =
+      [[FLTThreadSafeFlutterResult alloc] initWithResult:^(id result){
+      }];
+
+  [_camera startVideoRecordingWithResult:result];
+
+  [self waitForExpectations:@[ audioSettingsExpectation, videoSettingsExpectation ] timeout:1];
+
+  [writerMock stopMocking];
+  [videoMock stopMocking];
+  [audioSessionMock stopMocking];
+  [captureConnectionMock stopMocking];
+  [captureVideoDataOutputMock stopMocking];
+  [audioSessionMock stopMocking];
+  [videoSessionMock stopMocking];
+  [inputMock stopMocking];
+  [deviceMock stopMocking];
```

Comment:
This is a very worrying about of class mocking, given how fragile class mocks make entire test suites. Why can't we used dependency injection?

/cc @hellohuanlin since I'd like context on why this passed initial review.

Reply:
At least removed [... stopMocking].

Regarding `testSettings_shouldPassConfigurationToCameraDeviceAndWriter`: this is based on `testDidOutputSampleBufferIgnoreAudioSamplesBeforeVideoSamples` and `FLTCreateCamWithCaptureSessionQueue`. 
This test traces all fps and bitrate variables usages until they will appear as args in [AVAssetWriterInput assetWriterInputWithMediaType]. AVFoundation creates and uses AVAssetWriterInput somewhere at sometime on its own and I just `nice mock` one its method it to intercept correct usage of parameters.

What else kind of dependency injection you mean? Am i missed some example in camera_avfoundation?

Reply:
Ok, thank you for answer. Lack of first class DI ObjC framework in flutter/packages makes OCMock so sweet...

Pushed variant with DI. No mocking in `testSettings_shouldPassConfigurationToCameraDeviceAndWriter`. But a lot of hand code. And naming may be a subject.

PS: There is still reused `FLTCreateCamWithCaptureSessionQueueAndProvider` by @hellohuanlin with ~20 stubs spreading over all FLTCam.m. But should MediaSettings patch refactor all `camera_avfoundation`?

Sorry, I reaching 250 commits limit in this branch and have only force-push option on github because of Google CLA...